### PR TITLE
[Debugger] Make sure to lock around BreakpointStore when using it

### PR DESCRIPTION
**Pull request by jstedfast**
Might fix https://devdiv.visualstudio.com/DevDiv/_workitems/edit/935168/

### CODE REVIEW

File: main/src/addins/MonoDevelop.Debugger/MonoDevelop.Debugger.VSTextView/BreakpointManager.cs [hunk 1]
```diff
@@ -63,13 +63,21 @@ class ManagerBreakpoint
...
+			var breakpointStore = DebuggingService.Breakpoints;
 			var snapshot = textBuffer.CurrentSnapshot;
+			var bps = new List<Breakpoint> ();
 			var needsUpdate = false;
 
-			foreach (var breakpoint in DebuggingService.Breakpoints.GetBreakpointsAtFile (textDocument.FilePath)) {
-				if (breakpoint.Line > snapshot.LineCount)
-					continue;
+			lock (breakpointStore) {
+				foreach (var breakpoint in breakpointStore.GetBreakpointsAtFile (textDocument.FilePath)) {
```
Comment (Therzok) [inline main/src/addins/MonoDevelop.Debugger/MonoDevelop.Debugger.VSTextView/BreakpointManager.cs]: we null check GetBreakpointsAtFile somewhere else. Should this be null checked too?

**Comment by sgmunn** [comment]
@monojenkins backport release-8.2

File: main/src/addins/MonoDevelop.Debugger/MonoDevelop.Debugger.VSTextView/BreakpointManager.cs (hunk 1 above)
Comment (jstedfast) [reply main/src/addins/MonoDevelop.Debugger/MonoDevelop.Debugger.VSTextView/BreakpointManager.cs]: The result? Or the Breakpoints property? Neither should ever be null afaict.

File: main/src/addins/MonoDevelop.SourceEditor2/MonoDevelop.SourceEditor.QuickTasks/QuickTaskOverviewMode.cs [hunk 2]
```diff
@@ -723,16 +723,22 @@ protected void DrawCaret (Cairo.Context cr, double displayScale)
...
+			var breakpointStore = parentStrip.SourceEditorView.Breakpoints;
 
-				cr.SetSourceColor (SyntaxHighlightingService.GetColor (TextEditor.EditorTheme, EditorThemeColors.BreakpointMarker));
-				int r = 4;
-				cr.Rectangle (0, y  - r / 2, r, r);
-				cr.Fill ();
+			lock (breakpointStore) {
+				var breakPoints = breakpointStore.GetBreakpointsAtFile (TextEditor.FileName);
+
+				if (breakPoints == null)
```
Comment (Therzok) [inline main/src/addins/MonoDevelop.SourceEditor2/MonoDevelop.SourceEditor.QuickTasks/QuickTaskOverviewMode.cs]: Here is where we null checked it @jstedfast

Comment (jstedfast) [reply main/src/addins/MonoDevelop.SourceEditor2/MonoDevelop.SourceEditor.QuickTasks/QuickTaskOverviewMode.cs]: That seems unnecessary... all return paths return a new list and never null.

**Pull request merged**